src/Components/Main/TodoMain.tsx

```diff
@@ -0,0 +1,146 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import cn from 'classnames';
+
+type Props = {
+  todo: Todo;
+  processing: number;
+  todosQuantity: number;
+  setTodosQuantity: (todosQuantity: number) => void;
+  onDelete: (todoId: number) => Promise<void>;
+  onUpdate: (redactedTodo: Todo) => Promise<void>;
+  isEditing: number | undefined;
+  setIsEditing: (isEditing: number | undefined) => void;
+};
+
+export const TodoMain: React.FC<Props> = ({
+  todo,
+  processing,
+  onDelete,
+  onUpdate,
+  todosQuantity,
+  setTodosQuantity,
+  isEditing,
+  setIsEditing,
+}) => {
+  const [newTitle, setNewTitle] = useState(todo.title);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleChangeCompleted = () => {
+    onUpdate({
+      ...todo,
+      completed: todo.completed ? false : true,
```

Comment:
```suggestion
      completed: todo.completed,
```

can be a bit simplify